tests/unit-mpi/test_base.py

```diff
@@ -0,0 +1,146 @@
+                },
+                "compartment_i": {
+                    "type": "compartment",
+                    "cells": "Mosaic",
+                    "variable_name": "i_membrane",
+                    "sections": "all",
+                    "dt": 1,
+                    "start_time": 0.0,
+                    "end_time": 40.0,
+                },
+            },
+        }
+    },
+    {
+        "simconfig_fixture": "ringtest_baseconfig",
+        "extra_config": {
+            "target_simulator": "CORENEURON",
+            "inputs": {
+                "Stimulus": {
+                    "module": "pulse",
+                    "input_type": "current_clamp",
+                    "delay": 5,
+                    "duration": 50,
+                    "node_set": "RingA",
+                    "represents_physical_electrode": True,
+                    "amp_start": 10,
+                    "width": 1,
+                    "frequency": 50
+                }
+            },
+            "reports": {
+                "soma_v": {
+                    "type": "compartment",
+                    "cells": "Mosaic",
+                    "variable_name": "v",
+                    "sections": "soma",
+                    "dt": 0.1,
+                    "start_time": 0.0,
+                    "end_time": 18.0,
+                },
+                "compartment_i": {
+                    "type": "compartment",
+                    "cells": "Mosaic",
+                    "variable_name": "i_membrane",
+                    "sections": "all",
+                    "dt": 1,
+                    "start_time": 0.0,
+                    "end_time": 40.0,
+                },
+            },
+        }
+    }
+], indirect=True)
+@pytest.mark.mpi(ranks=2)
+def test_neurodamus_with_neuron_and_coreneuron(create_tmp_simulation_config_file, mpi_ranks):
+    """Test Neurodamus/neuron with and without MPI."""
+    from neurodamus import Neurodamus
+    from neurodamus.core import MPI as neuron_MPI
+
+    assert neuron_MPI.size == mpi_ranks == size == 2
+    assert neuron_MPI.rank == rank
+
+    n = Neurodamus(create_tmp_simulation_config_file, disable_reports=True)
+    local_gids_ref = [[1, 3], [2]]
+    local_gids = n.circuits.get_node_manager("RingA").local_nodes.final_gids()
+
+    np.testing.assert_allclose(local_gids, local_gids_ref[neuron_MPI.rank])
+    n.run()
+
+
+@pytest.mark.parametrize("create_tmp_simulation_config_file", [
+    {
+        "simconfig_fixture": "ringtest_baseconfig",
+        "extra_config": {
+            "target_simulator": "NEURON",
+            "node_set": "RingA_oneCell",
+        }
+    },
+    {
+        "simconfig_fixture": "ringtest_baseconfig",
+        "extra_config": {
+            "target_simulator": "CORENEURON",
+            "node_set": "RingA_oneCell",
+        }
+    }
+], indirect=True)
+@pytest.mark.mpi(ranks=2)
+def test_empty_rank_with_neuron_and_coreneuron(create_tmp_simulation_config_file, mpi_ranks):
```

Comment:
It is fine to test only with CoreNEURON. But you can also include NEURON.
In addition, can you also check the fake cell `dat` file existing in the `coreneuron_input` folder ? I guess the fake cell file should be named as `1002_1.dat`. You will need to enable the option `keep_build=True`.

Reply:
done. I noticed that each rank gets its own cwd. Thus, only rank 0 has the build folder. For save/restore it should still be ok because we can provide the restore path. What do you think?

I would not add an mpi test for save/restore in this pr. I think it goes out of scope... as you prefer tho. I can also add that if you think it should be in the same pr that introduces mpi to our framework

Reply:
Another question: why only rank0 writes? The fake cell files should be written in rank1.

Reply:
each rank gets a different folder for cwd. I noticed that in coreneuron, only rank 0 writes the folder `build`. The fake cell files are in the folder for rank 0

Reply:
it is a "feature" of pytest-isolate-mpi. I tried to remove it but I do not get the mpi or the forking in that case

Reply:
I checked and these are my conclusions (I am not 100% sure that there isn't something else afoot):
- neurodamus is importend. It gets cwd and sets `CoreNeuron.datadir`
- `pytest-isolate-mpi` changes cwd (one per rank) and starts the simulations
- all the ranks write in the folder for rank 0 (as coreneuron would want to do)
- if you assert if a file is there, it will do it in its own, changed, cwd. Thus we should check files only in rank 0

I will add this in the comments as it is interesting to retain